diff --git a/flake.nix b/flake.nix
index b275f4c669..d1db8b974a 100644
--- a/flake.nix
+++ b/flake.nix
@@ -37,7 +37,6 @@
         n2c.follows = "n2c";
         data-merge.follows = "data-merge";
         cardano-iohk-nix.follows = "iohk-nix";
-        cardano-node.follows = "cardano-node";
         cardano-db-sync.follows = "cardano-db-sync";
         cardano-wallet.follows = "cardano-wallet";
       };
@@ -50,11 +49,6 @@
     grafonnix.url = "github:divnix/grafonnix/auto-convert";
     # --------------------------------------------------------------
     # --- Bride Heads ----------------------------------------------
-    # TODO: remove cardano-node (and use self) when mono-repo branch is merged:
-    cardano-node = {
-      url = "github:input-output-hk/cardano-node/1.35.2";
-      flake = false;
-    };
     cardano-db-sync.url = "github:input-output-hk/cardano-db-sync/13.0.2";
     cardano-wallet.url = "github:input-output-hk/cardano-wallet/v2022-07-01";
     ogmios = {
diff --git a/nix/automation/devshells.nix b/nix/automation/devshells.nix
index e55834b276..7171255e8f 100644
--- a/nix/automation/devshells.nix
+++ b/nix/automation/devshells.nix
@@ -73,12 +73,6 @@ in {
       inputs.cells.cardano.devshellProfiles.world
     ];
   };
-  monorepo = std.lib.mkShell {
-    imports = [
-      cardanoWorld
-      inputs.cells.cardano.devshellProfiles.monorepo
-    ];
-  };
   minimal = std.lib.mkShell {
     imports = [
       cardanoWorld
diff --git a/nix/cardano/devshellProfiles.nix b/nix/cardano/devshellProfiles.nix
index f20934dd60..1515e39000 100644
--- a/nix/cardano/devshellProfiles.nix
+++ b/nix/cardano/devshellProfiles.nix
@@ -117,12 +117,4 @@ rec {
       }
     ];
   };
-  monorepo = _: {
-    commands = [
-      {
-        package = inputs.cells.cardano.prepare-mono-repo.merge-mono-repo;
-        category = "nix-build";
-      }
-    ];
-  };
 }
diff --git a/nix/cardano/packages/default.nix b/nix/cardano/packages/default.nix
index 7bd4dd27b7..90e32b0f57 100644
--- a/nix/cardano/packages/default.nix
+++ b/nix/cardano/packages/default.nix
@@ -3,7 +3,7 @@
 ,
 }:
 let
-  inherit (inputs) self std nixpkgs iohk-nix cardano-node
+  inherit (inputs) self std nixpkgs iohk-nix
     cardano-wallet cardano-db-sync ogmios cardano-graphql cardano-explorer-app nix-inclusive;
   inherit (inputs.cells) cardano;
   inherit (nixpkgs) lib;
@@ -27,8 +27,7 @@ let
     (import ./haskell.nix {
       inherit lib haskell-nix;
       inherit (inputs) byron-chain;
-      # TODO: switch to self after mono-repo branch is merged:
-      src = cardano-node;
+      src = self;
     }).extend (final: prev: {
       release = nixpkgs.callPackage ./binary-release.nix {
         inherit (final.pkgs) stdenv;
diff --git a/nix/cardano/packages/haskell.nix b/nix/cardano/packages/haskell.nix
index 430638aa22..6ee44039b7 100644
--- a/nix/cardano/packages/haskell.nix
+++ b/nix/cardano/packages/haskell.nix
@@ -42,14 +42,9 @@ in
       name = "cardano-world-src";
       filter = path: type:
         let relPath = lib.removePrefix "${src.outPath}/" path; in
-        # excludes directories not part of cabal project:
-        (type != "directory" || (builtins.match ".*/.*" relPath != null) || (!(lib.elem relPath [
-          "nix"
-          "doc"
-          "docs"
-        ]) && !(lib.hasPrefix "." relPath)))
-        # only keep cabal.project from files at root:
-        && (type == "directory" || builtins.match ".*/.*" relPath != null || (relPath == "cabal.project"))
+        # only keep cabal.project and directories under src:
+        (relPath == "cabal.project" || relPath == "src" || (type == "directory" && (builtins.match "src/.*" relPath != null))
+          || (builtins.match "src/.*/.*" relPath != null))
         && (lib.cleanSourceFilter path type)
         && (haskell-nix.haskellSourceFilter path type)
         && !(lib.hasSuffix ".gitignore" relPath)
