diff --git a/nix/automation/devshells.nix b/nix/automation/devshells.nix
index e55834b276..7171255e8f 100644
--- a/nix/automation/devshells.nix
+++ b/nix/automation/devshells.nix
@@ -73,12 +73,6 @@ in {
       inputs.cells.cardano.devshellProfiles.world
     ];
   };
-  monorepo = dev.mkShell {
-    imports = [
-      cardanoWorld
-      inputs.cells.cardano.devshellProfiles.monorepo
-    ];
-  };
   minimal = dev.mkShell {
     imports = [
       cardanoWorld
diff --git a/nix/cardano/devshellProfiles.nix b/nix/cardano/devshellProfiles.nix
index f20934dd60..1515e39000 100644
--- a/nix/cardano/devshellProfiles.nix
+++ b/nix/cardano/devshellProfiles.nix
@@ -117,12 +117,4 @@ rec {
       }
     ];
   };
-  monorepo = _: {
-    commands = [
-      {
-        package = inputs.cells.cardano.prepare-mono-repo.merge-mono-repo;
-        category = "nix-build";
-      }
-    ];
-  };
 }
